### PR TITLE
add autosmooth documentation

### PR DESCRIPTION
**Pull request by byassine93**
### What does this PR do?

Adds a description to the autosmooth function created in prod (enabled only for Datadog for now).

### Motivation

[Release of the algorithm to prod](https://trello.com/c/w5aTMqr9/2845-implement-asap-in-dogweb)
[Trello Card ](https://trello.com/c/clqwJi4C/2872-blocked-on-auto-smoother-release-add-auto-smoother-to-documentation)

### Preview link

https://docs-staging.datadoghq.com/yassine.benazzou/add-autosmooth-documentation/graphing/miscellaneous/functions/#autosmooth

### CODE REVIEW

File: content/graphing/miscellaneous/functions.md [hunk 1]
```diff
@@ -287,6 +287,18 @@ Delta value between points for a given metric
 
 ## Smoothing
 
+### autosmooth()
+
+Automatically removes noise while preserving the trend of the time series.
+
+The `autosmooth()` function applies a moving average with an automatically selected span. It smoothes a time series while preserving its trend. In this example, the function chooses the optimal span to smooth the time series:
+
+{{< img src="graphing/miscellaneous/functions/autosmooth_illustration.png" alt="autosmooth illustration" responsive="true" style="width:80%;">}}
```
Comment (StephenKappel) [inline content/graphing/miscellaneous/functions.md]: does this image file need to be added to the PR?

File: content/graphing/miscellaneous/functions.md [hunk 2]
```diff
@@ -287,6 +287,18 @@ Delta value between points for a given metric
...
+
+When used on a `group by` query (e.g. `avg by`), the same span is applied on all the time series. If used on several metrics in the same graph, different spans could be selected to optimally smooth each one of the metric time series.
+
+The algorithm is inspired by the [ASAP algorithm][8] - you can read more about it in this blog post.
```
Comment (StephenKappel) [inline content/graphing/miscellaneous/functions.md]: nit: can we put the link on the words "this blog post".

Comment (byassine93) [reply content/graphing/miscellaneous/functions.md]: We don't have the link yet. I was going to add it afterwards. The trello card is blocked by that as well

Comment (StephenKappel) [reply content/graphing/miscellaneous/functions.md]: I misunderstood originally. This makes more sense now!

**Comment by l0k0ms** [comment]
We refactored heavily the functions documentation. This PR content is now here: https://github.com/DataDog/documentation/pull/3016

**Pull request closed without merging**